chain/deals/provider_utils.go

```diff
@@ -125,3 +136,81 @@ func (p *Provider) getWorker(miner address.Address) (address.Address, error) {
 
 	return address.NewFromBytes(r.Return)
 }
+
+var _ datatransfer.RequestValidator = &ProviderRequestValidator{}
+
+// ProviderRequestValidator validates data transfer requests for the provider
+// in a storage market
+type ProviderRequestValidator struct {
+	deals *statestore.StateStore
+}
+
+// RegisterProviderValidator is an initialization hook that registers the provider
+// request validator with the data transfer module as the validator for
+// StorageDataTransferVoucher types
+func RegisterProviderValidator(lc fx.Lifecycle, mrv *ProviderRequestValidator, dtm dtypes.ProviderDataTransfer) {
+	lc.Append(fx.Hook{
```

Comment:
Wait, why does this need to be called OnStart? Can't it be done in the constructor? does it spin up a goroutine?

Reply:
It could be in the constructor I guess. The validator doesn't otherwise contain any reference to the DT module. This also provides a way to make sure the validator gets instantiated. (cause no other modules depend on it ATM).

Reply:
I'm under the impression (perhaps incorrect) that FX doesn't instantiate modules if they aren't depended on something that's in a lifecycle hook or instantiated directly by top level code.

Reply:
@magik6k has been working with FX a lot longer than me so perhaps he can speak to this.